File: 1528-shuffle-string/1528-shuffle-string.js
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
    const res = Array(s.length)
    for (let i=0; i < indices.length; i++) {
        res[indices[i]] = s[i]
    }
    return res.join("")
};

restoreString = function(s, indices) {
    return indices.map((idx, i) => ([idx, s[i]])).sort(([a], [b]) => a - b).map(([,a]) => a).join("")
}

File: 1528-shuffle-string/1528-shuffle-string.kt
class Solution {
    fun restoreString(s: String, indices: IntArray): String {
        val res = CharArray(s.length)
        for (i in 0..(indices.size-1)) {
            res[indices[i]] = s[i]
        }
        return res.joinToString(separator="")
    }
}